Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py

```diff
@@ -138,11 +138,25 @@ def main(cls, args, repository, why=None, redact=False, **kwargs):
             else:
                 log.warning("'{}' has no spaces, assuming user intends it to be a branch name".format(args.issue))
 
-        if issue:
-            for tracker in Tracker._trackers:
-                if isinstance(tracker, radar.Tracker) and tracker.radarclient():
-                    issue.cc_radar(block=True)
-                    break
+        needs_radar = issue and not isinstance(issue.tracker, radar.Tracker)
+        needs_radar = needs_radar and any([
+            isinstance(tracker, radar.Tracker) and tracker.radarclient()
+            for tracker in Tracker._trackers
+        ])
+        needs_radar = needs_radar and not any([
+            isinstance(reference.tracker, radar.Tracker)
+            for reference in issue.references
+        ])
+
+        if needs_radar:
+            rdar = None
+            if not getattr(args, 'defaults', None):
+                sys.stdout.write('Existing radar to CC (leave empty to create new radar)')
+                sys.stdout.flush()
+                rdar = Tracker.from_string(Terminal.input(': '))
```

Comment:
Everything except for the first one will work here, and `Tracker.from_string` will return `None` if it doesn't match. I can make the `NNNN` approach work, the only reason it doesn't is because it's ambiguous and might not refer to radar, in general, but that's not true in this particular piece of code.

Reply:
Following up more specifically, [webkitbugspy/radar.py](https://github.com/WebKit/WebKit/blob/a6840766432adbc9769019526c502c99476e8696/Tools/Scripts/libraries/webkitbugspy/webkitbugspy/radar.py#L42) show you what we recognize as a radar string.

Reply:
Seems reasonable. I think I'm a little burned from back when we were using `webkit-patch`, and that _only_ accepted a bug number sans URL. But that memory will fade away :)